perfzero/lib/cloud_manager.py

```diff
@@ -133,9 +134,37 @@ def create(username, project, zone, machine_type, accelerator_count,
   logging.info('Successfully created gcloud computing instance %s with %s accelerator.\n',  # pylint: disable=line-too-long
                instance_name, accelerator_count)
 
+  # Wait until we can ssh to the newly created computing instance
+  cmd = 'gcloud compute ssh {} --project={} --zone={} --strict-host-key-checking=no --command="ls"'.format(instance_name, project, zone)  # pylint: disable=line-too-long
+  ssh_remaining_retries = 12
+  ssh_error = None
+  while ssh_remaining_retries > 0:
+    ssh_remaining_retries -= 1
+    try:
+      run_command(cmd, is_from_user=False)
+      ssh_error = None
+    except Exception as error:  # pylint: disable=W0703
+      ssh_error = error
+      if ssh_remaining_retries:
+        logging.info('Cannot ssh to the computing instance. Try again after 5 seconds')  # pylint: disable=line-too-long
+        time.sleep(5)
+      else:
+        logging.error('Cannot ssh to the computing instance after 60 seconds due to error:\n%s', str(ssh_error))  # pylint: disable=line-too-long
+
   cmd = 'gcloud compute ssh {} --project={} --zone={} --command="git clone {}"'.format(
       instance_name, project, zone, 'https://github.com/tensorflow/benchmarks.git')  # pylint: disable=line-too-long
-  logging.info('Run the command below to checkout PerfZero code on the computing instance:\n%s\n', cmd)  # pylint: disable=line-too-long
+  if not ssh_error:
+    run_command(cmd, is_from_user=True)
+    logging.info('Successfully checked-out PerfZero code on the computing instance\n')  # pylint: disable=line-too-long
+  else:
+    logging.info('Run the command below to checkout PerfZero code on the computing instance:\n%s\n', cmd)  # pylint: disable=line-too-long
+
+  cmd = 'gcloud compute ssh {} --project={} --zone={} --command="sudo usermod -a -G docker $USER"'.format(instance_name, project, zone)  # pylint: disable=line-too-long
```

Comment:
Can we run these commands together? The script is not catching some command specific errors so this part of code seems redundant to me.

Reply:
Hmm... do  you mean that we should execute `run_command(...)` only once and put all the commands, concatenated by `&&`, as parameter to `run_command(...)`?

My concern with this approach is that, say we do `A && B && C`, if command `B` exists with non-zero exit code, `run_command` will not know which part failed, and `cloud_manager.py` will not be able to clearly show what has succeeded and what has failed. This leaves the state of the computing instance in a state that is not known to user.

In comparison, currently `cloud_manager.py` does catch error per command and show INFO message per command, e.g. `successfully checked out PerfZero code...`. This seems more user friendly, right?

Reply:
```
def try_to_run_command(cmd, success_msg, error_msg) {
  if ssh_error:
  else:
}
```

Reply:
Thanks! I have updated the code to remove the logic duplication.